build.gradle

```diff
@@ -142,75 +109,118 @@ wrapper {
+    rejectVersionIf {
+        isNonStable(it.candidate.version)
+    }
 }
 
+def appVersion() {
+    return rootProject.version
+            .replaceAll("\\-", ".")
+            .replaceAll("alpha", "0")
+            .replaceAll("beta", "0")
+            .replaceAll("release", "0")
+}
+
+task buildDebPackage(type: Exec) {
+    commandLine jpackage(),
+            '--verbose',
+            '--type', 'deb',
+            '--name', appName,
+            '--description', rootProject.description,
+            '--vendor', vendor,
+            '--license-file', licenceFile,
+            '--icon', './native/rails.png',
+            '--file-associations', './native/rails-linux-association.properties',
+            '--app-version', appVersion(),
+            '--input', 'build/libs',
+            '--main-jar', shadowJar.archiveName,
+            '--linux-package-name', 'rails-18xx',
+            '--linux-app-category', 'games',
+            '--linux-shortcut'
+}
+
+task buildRpmPackage(type: Exec) {
+    commandLine jpackage(),
+            '--verbose',
+            '--type', 'rpm',
+            '--name', appName,
+            '--description', rootProject.description,
+            '--vendor', vendor,
+            '--license-file', licenceFile,
+            '--icon', './native/rails.png',
+            '--file-associations', './native/rails-linux-association.properties',
+            '--app-version', appVersion(),
+            '--input', 'build/libs',
+            '--main-jar', shadowJar.archiveName,
+            '--linux-package-name', 'rails-18xx',
+            '--linux-app-category', 'games',
+            '--linux-shortcut'
+}
+
+task buildExePackage(type: Exec) {
+    commandLine jpackage(),
+            '--verbose',
+            '--type', 'exe',
+            '--name', appName,
+            '--description', rootProject.description,
+            '--vendor', vendor,
+            '--license-file', licenceFile,
+            '--icon', './native/rails.ico',
+            '--file-associations', './native/rails-windows-association.properties',
+            '--app-version', appVersion(),
+            '--input', 'build/libs',
+            '--main-jar', shadowJar.archiveName,
+            '--win-dir-chooser',
+            '--win-per-user-install',
+            '--win-menu',
+            '--win-shortcut'
+}
+
+task buildDmgPackage(type: Exec) {
+    commandLine jpackage(),
+            '--verbose',
+            '--type', 'dmg',
+            '--name', appName,
+            '--description', rootProject.description,
+            '--vendor', vendor,
+            '--license-file', licenceFile,
+            '--icon', './native/rails.ico',
+            '--file-associations', './native/rails-osx-association.properties',
+            '--app-version', appVersion(),
+            '--input', 'build/libs',
+            '--main-jar', shadowJar.archiveName,
+            '--mac-package-name', 'rails-18xx'
+}
+
+task renameExecutables(type: Copy) {
+    from('.') {
```

Comment:
Would suggest we instead set a destination path in the jpackage builds (maybe build/jpackage or something in build) and then limit this `from` to that directory.

Reply:
Let us do that in a later PR